### ISSUE TITLE
File Error even if file does not exist

### ISSUE THREAD

**Issue by enoch85** [owncloud/ios#622]
(Btw, you should add a issue_template.md)

### Expected behaviour
File should be uploaded
### Actual Behaviour
File doesn't get uploaded and in Failed uploads it says it's due to "File error", even if the file *does not* exist in the upload dir (instant upload) Similar to #556
### Steps to reproduce
1. Take a photo
2. Wait for instant upload to be triggered, or trigger it manually.

### Logs
**owncloud.log**

```
Fatal	webdav	Exception: {"Message":"HTTP\/1.1 405 The resource you tried to create already exists","Exception":"Sabre\\DAV\\Exception\\MethodNotAllowed","Code":0,"Trace":"#0 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(590): Sabre\\DAV\\Server->createCollection('InstantUpload', Object(Sabre\\DAV\\MkCol))\n#1 [internal function]: Sabre\\DAV\\CorePlugin->httpMkcol(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#2 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#3 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\\Event\\EventEmitter->emit('method:MKCOL', Array)\n#4 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#5 \/var\/www\/owncloud\/apps\/dav\/appinfo\/v1\/webdav.php(55): Sabre\\DAV\\Server->exec()\n#6 \/var\/www\/owncloud\/remote.php(138): require_once('\/var\/www\/ownclo...')\n#7 {main}","File":"\/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php","Line":1121,"User":"*****"}
```

**Actual upload folder**

![deepinscreenshot20160405222753](https://cloud.githubusercontent.com/assets/4511254/14297242/2380823c-fb7e-11e5-9806-c833d41c1c2f.jpg)

### Server
ownCloud 9.0
Ubuntu 14.04
PHP 7
Apache 2.4

### Client
iOS 9.3.1
App: 3.4.8

**Comment by enoch85** [comment]
cc @DeepDiver1975 as this issue may be in Core as well.

**Comment by nasli** [comment]
@enoch85, thanks for your feedback!
Don´t worry for this message. This issue is duplicated check https://github.com/owncloud/ios/issues/373#issuecomment-114029563, as ggdiez commented, we are not checking that.

**Comment by enoch85** [comment]
But, the file doesn't get uploaded, isn't that an issue for the iOS team?

**Comment by nasli** [comment]
Yes, it is. I thought this issue was about only the error, and in this one https://github.com/owncloud/ios/issues/556#issuecomment-205979085 the failed upload, it is the same, it is? could you reopen it? Will check it but here we could not reproduce so far. It happens always for you?

**Comment by enoch85** [comment]
@nasli The error I posted in #556 is related to this issue, and yes I still see it - but that issue was about something else. I thought it would be better to track it here instead. 

It's hard to reproduce, and it doesn't happen all the time. That's why it's so hard to debug. Is there any logs on the iOS that I can extract?

**Comment by enoch85** [comment]
Maybe the file gets uploaded to the database, but not to the data folder? The reason I think this is that it's like the file is already existing `"HTTP\/1.1 405 The resource you tried to create already exists"`, but looking in the data folder it doesn't.

**Comment by thomasgg23** [comment]
Same problem for me. But not all the time. 
Got the same message "HTTP\/1.1 405 The resource you tried to create already exists" in the logs and no pictures are uploaded.

Server
ownCloud 9.0.1
Ubuntu 14.04.4
PHP 5.5.9
Apache 2.4.7
Selfsigned certificate and different Server Port

Client
iOS 9.3.1
App: 3.4.8

**Comment by StatLaw** [comment]
Both of you are using iOS 9.3.1. Any way you can check whether this issue exists under 9.2.1? I've posted on what seems to be a different issue, but perhaps they are related. Do your server logs show that the owncloud app issued an appropriate PUT?

**Comment by SweBrov** [comment]
I observed the issue on 9.2.1 as well.

**Comment by enoch85** [comment]
@nasli Could you please prioritize this? :)

**Comment by nasli** [comment]
Hi @enoch85, instant uploads and uploads inside the app are currently under a refactor process in order to use the Photo framework instead of Assets Library Framework that is available since iOS8. This framework will give us more options and notifications events.

Let's track if with this changes fixed also the issue. ;)

**Comment by enoch85** [comment]
https://github.com/owncloud/ios/pull/666 & https://github.com/owncloud/ios/pull/679

**Comment by nasli** [comment]
@enoch85 could you try with new version 3.5.0, if needed reopen thanks!

**Comment by Flexxxo** [comment]
iOS Client 3.5.0
iOS 9.3.5
OC 8.2.7

Still "The resource you tried to create already exists" webdav errors in the web administration and the queue of "Instant Uploads" on iPad / iPhone getting stuck.

**Comment by enoch85** [comment]
Still got the file error stuff, but I have not investigated further.
On vaccation now ☀☀☀☀
Best regards
Daniel Hansson
www.techandme.se
Sent from Nexus 5x, sorry for typos.
Den 2 sep. 2016 4:55 fm skrev Noelia Alvarez <notifications@github.com>:@enoch85 could you try with new version 3.5.0, if needed reopen thanks!

**Comment by nasli** [comment]
Thanks for your feedback @Flexxxo,
some more question to dig in the problem,
What kind of error appears on the uploads queue? File error, conflict error ?
What's your device model?
How much free space you have on your device?
All images fail or some of them? 
What size is the image that fails?
Thx

**Comment by Flexxxo** [comment]
Hello nasli,

iPad Air 1, 64 GB device with about 50 free GB of space.

The queue is just stalled. Nothing happens.

When I close the app using double tap on Home button and swipe it away it uploads 1-2 files when started again. Then it stops.

File size deos not seem to matter as the queue lists Instant Uploads between 20 kb und 1.5 mb.

Some of the files show "Waiting for server connection", some show "Waiting for upload" but the upload seems frozen. No progress until I restart the app.

I tested with the two sliders "Instant Upload" and "Instant Upload in Background" switched on and off and alss using cellular (4G) and/or WiFi (DSL connection with 120 mbit/s) - for 30 files in the queue I had to close and restart the app multiple times.

Sometimes it is enough to use the Home button to loose focus on the app and opening it again from the memory to upload 1-2 files again.

Then the upload stops again and will not continue without my interaction, no matter how long I am waiting.

Basically upload in Background (which would mean in my terms: OC begins uploading after e.g. a photo was taken) never really worked. I also have to open the app - no matter if a PIN is set or not to get the upload queue to be worked on.

The queue shows no errors, when this happens.

The same thing happens in my iPhone 6 - exactly the same behaviour of the app.

**Comment by enoch85** [comment]
I can confirm that the error still presists. This time the photo didn't get uploaded due to "file error" (null@domain.com) and when I checked the local space the photo/file is there, but I can't retry it manually. 

On Android that's possible, and maybe you should consider adding that option as well instead if just "delete"...

Still on vaccation so can't provide logs and such. Still same specs as in original post.

**Comment by enoch85** [comment]
Let's track this in #761 instead.
